draft-ietf-acme-acme.md

```diff
@@ -982,65 +982,73 @@ A client may wish to change the public key that is associated with a
 registration in order to recover from a key compromise or proactively mitigate
 the impact of an unnoticed key compromise.
 
-To change the key associated with an account, the client sends a POST request
-containing a key-change object with the following fields:
+To change the key associate with an account, the client first constructs a
+key-change object describing the change that it would like the server to make:
 
 oldKey (required, JWK):
-: The JWK representation of the original key (i.e., the client's current account
+: The JWK thumbprint of the original key (i.e., the client's current account
 key)
 
 newKey (requrired, JWK):
-: The JWK representation of the new key
+: The JWK thumbprint of the new key
 
-The JWS of this POST must have two signatures: one signature from the existing
-key on the account, and one signature from the new key that the client proposes
-to use. This demonstrates that the client actually has control of the
-private key corresponding to the new public key. The protected header must
-contain a JWK field containing the current account key.
+The client then encapsulates the key-change object in a JWS, signed with the
+client's current account key (i.e., the key matching the "oldKey" value).  This
+JWS is NOT REQUIRED to have the "nonce" and "url" fields, despite the general
+requirement that these fields be present.
```

Comment:
It seems like design-wise, there are two real alternatives:

1. Same requirements as the outer JWS
2. No ACME-specific requirements

I had in mind (2) here because it seemed like a hassle to burn two nonces at once.  Also, the benefit of the "url" field here seems pretty small, since (at least in the context of ACME), this object can only be replayed by the holder of the old account key, and doesn't isn't useful for anything other than rolling a key.

But hey, rigidity is good and nonces are plentiful, so why not (1).  You also get the benefit of being able to always require `nonce` and `url` in your `sign()` method.  The main cost is going to be the need for a more nonce-priming HEAD requests, but since this is an infrequent operation, that doesn't seem tragic.